python/extensions.bzl

```diff
@@ -127,3 +152,74 @@ pip = module_extension(
         "parse": tag_class(attrs = _pip_parse_ext_attrs()),
     },
 )
+
+# This function allows us to build the label name of a label
+# that is not passed into the current context.
+# The module_label is the key element that is passed in.
+# This value provides the root location of the labels
+def _repo_mapped_label(module_label, sub_module, apparent):
+    return Label("@@{module}~{sub_module}~{apparent}".format(
+        module = module_label.workspace_name,
+        sub_module = sub_module,
+        apparent = apparent,
+    ))
+
+# We are doing some bazel stuff here that could use an explanation.
+# The basis of this function is that we need to create a symlink to
+# the python binary that exists in a different repo that we know is
+# setup by rules_python.
+#
+# We are building a Label like
+# @@rules_python~override~python~python3_x86_64-unknown-linux-gnu//:python
+# and then the function creates a symlink named python to that Label.
+#
+# We are building the Label name dynamically,  and can do this even
+# though the Label is not passed into this function.  If we choose
+# not do this a user would have to write another 16 lines
+# of configuration code, but we are able to save them that work
+# because we know how rules_python works internally.  We are using
+# functions from private:toolchains_repo.bzl which is where the repo
+# is being built. The repo name differs between host OS and platforms
+# and the functions from toolchains_repo gives us this functions that
+# information.
+def _host_hub_impl(repo_ctx):
+    repo_ctx.file("BUILD.bazel", "")
+
+    # The two get_ functions we use are also utilized when building
+    # the repositories for the different interpreters.
+    (os, arch) = get_host_os_arch(repo_ctx)
+    host_platform = "{}_{}//:python".format(
+        repo_ctx.attr.extension_name,
+        get_host_platform(os, arch),
+    )
+
+    # the attribute is set to attr.label(default = "//:_"), which
+    # provides us the root module name.
+    # The sub_module name "python" is created by the rules_python
+    # folder structure.  Ergo rules_python/python.
+    # We then have the OS and platform specific name of the python
+    # interpreter.
+    label = _repo_mapped_label(repo_ctx.attr._module, "python", host_platform)
+
+    # create the symlink in order to set the interpreter for pip.
+    repo_ctx.symlink(label, "python")
+
+# This repository rule finds the host's Python interpreter and creates a symlink to it.
+# We use this rule to set the pips interpreter target when using different Operating
+# Systems with the same project.
+# This code exists because of changes bzlmod does not allow the use
+# of native.load_toolchain.
```

Comment:
Please remove; this isn't because of native.register_toolchain, but because of repo remapping and visibility

Reply:
Please example more because I started trying to do that call before all this work.

Reply:
@rickeylev addressed